mlflow/langchain/__init__.py

```diff
@@ -94,6 +98,19 @@ def get_default_conda_env():
     return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
 
 
+def get_retriever_pyfunc_model(fn):
+    class Retriever(mlflow.pyfunc.PythonModel):
+        def load_context(self, context):
+            local_dir = mlflow.artifacts.download_artifacts(context.artifacts[PERSIST_DIR_KEY])
+            self.retriever = fn(local_dir)
+
+        def predict(self, context, model_input):
+            # This is a hacky way to expose the retriever object.
+            return self.retriever
```

Comment:
Seems like we can just run `fn(mlflow.artifacts.download_artifacts(context.artifacts[PERSIST_DIR_KEY]))` in the `_load_model()` method, right? 

Would rather avoid packaging up a custom pyfunc just for this purpose

Reply:
Thanks for the suggestion. I'm not sure whether `fn(mlflow.artifacts.download_artifacts(context.artifacts[PERSIST_DIR_KEY]))` would be sufficient to pickle the `fn`. Let me try it.